src/test/suite/grammar.test.ts

```diff
@@ -0,0 +1,78 @@
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const vsctm = require('vscode-textmate');
+const oniguruma = require('vscode-oniguruma');
+
+/**
+ * Utility to read a file as a promise
+ */
+function readFile(path) {
+    return new Promise((resolve, reject) => {
+        fs.readFile(path, (error, data) => error ? reject(error) : resolve(data));
+    })
+}
+
+const wasmBin = fs.readFileSync(path.join(__dirname, '../../../node_modules/vscode-oniguruma/release/onig.wasm')).buffer;
+const vscodeOnigurumaLib = oniguruma.loadWASM(wasmBin).then(() => {
+    return {
+        createOnigScanner(patterns) { return new oniguruma.OnigScanner(patterns); },
+        createOnigString(s) { return new oniguruma.OnigString(s); }
+    };
+});
+
+// Create a registry that can create a grammar from a scope name.
+const registry = new vsctm.Registry({
+    onigLib: vscodeOnigurumaLib,
+    loadGrammar: (scopeName) => {
+        if (scopeName === 'source.atf') {
+            const grammar_path = path.join(__dirname, '../../../syntaxes/atf.tmLanguage.json')
+            return readFile(grammar_path).then(data => vsctm.parseRawGrammar(data.toString(), grammar_path))
+        }
+        console.log(`Unknown scope name: ${scopeName}`);
+        return null;
+    }
+});
+
+suite('Grammar Test Suite', () => {
+    vscode.window.showInformationMessage('Start all tests.');
+
+    test('Tokenization test', async () => {
+
+        // Load the ATF grammar and any other grammars included by it async.
+        const grammar = await registry.loadGrammar('source.atf');
+
+        const text = fs.readFileSync(path.join(__dirname, '../../../src/test/suite/example.atf')).toString().split(os.EOL)
+        const expected_scopes = [
+            ['keyword.other.ampersand.atf'],
+            ['support.class.hash.atf', 'source.atf'],
+            ['support.class.hash.atf', 'source.atf'],
+            ['support.class.hash.atf', 'source.atf'],
+            ['support.class.hash.atf', 'source.atf'],
+            ['keyword.control.at.atf'],
+            ['keyword.control.at.atf'],
+            ['variable.language.text.atf', 'source.atf'],
+            ['support.class.hash.atf', 'source.atf'],
+            ['keyword.control.at.atf', 'source.atf'],
+            ['keyword.control.at.atf'],
+            ['source.atf'],
+            ['variable.language.text.atf', 'source.atf'],
+            ['source.atf']
+        ]
+
+        let ruleStack = vsctm.INITIAL;
+        for (let i = 0; i < text.length; i++) {
+            const line = text[i];
+            const lineTokens = grammar.tokenizeLine(line, ruleStack);
+            for (let j = 0; j < lineTokens.tokens.length; j++) {
+                const token = lineTokens.tokens[j];
+                assert.equal(token.scopes[token.scopes.length-1], expected_scopes[i][j]);
```

Comment:
This is [deprecated](https://nodejs.org/api/assert.html#assert_assert_equal_actual_expected_message), use `assert.strictEqual` instead (which is what it calls).

Reply:
Ok, done!